internal/github/publish_github_gist_test.go

```diff
@@ -216,3 +218,52 @@ func (s *GithubPublishGistTestSuite) TestResetOldFiles() {
 	})
 	s.Require().Nil(diags)
 }
+
+func (s *GithubPublishGistTestSuite) TestError() {
+	s.cli.On("Gists").Return(s.gistCli)
+	s.gistCli.On("Create", mock.Anything, mock.Anything).Return(nil, nil, errors.New("github api failed"))
+	ctx := context.Background()
+	titleMeta := plugindata.Map{
+		"provider": plugindata.String("title"),
+		"plugin":   plugindata.String("blackstork/builtin"),
+	}
+
+	diags := s.plugin.Publish(ctx, "github_gist", &plugin.PublishParams{
+		Config: plugintest.NewTestDecoder(s.T(), s.plugin.Publishers["github_gist"].Config).
+			SetAttr("github_token", cty.StringVal("testtoken")).
+			Decode(),
+		Args: plugintest.NewTestDecoder(s.T(), s.plugin.Publishers["github_gist"].Args).
+			SetAttr("description", cty.StringVal("test description")).
+			SetAttr("filename", cty.StringVal("filename.md")).
+			Decode(),
+		Format: plugin.OutputFormatMD,
+		DataContext: plugindata.Map{
+			"document": plugindata.Map{
+				"meta": plugindata.Map{
+					"name": plugindata.String("test_document"),
+				},
+				"content": plugindata.Map{
+					"type": plugindata.String("section"),
+					"children": plugindata.List{
+						plugindata.Map{
+							"type":     plugindata.String("element"),
+							"markdown": plugindata.String("# Header 1"),
+							"meta":     titleMeta,
+						},
+						plugindata.Map{
+							"type":     plugindata.String("element"),
+							"markdown": plugindata.String("Lorem ipsum dolor sit amet, consectetur adipiscing elit."),
+						},
+					},
+				},
+			},
+		},
+		DocumentName: "test_doc",
+	})
+	s.Require().Len(diags, 1)
+	diagtest.Asserts{{
+		diagtest.IsError,
+		diagtest.SummaryEquals("Failed to create gist"),
+		diagtest.DetailEquals("github api failed"),
+	}}.AssertMatch(s.T(), diags, nil)
```

Comment:
```suggestion
	diagtest.Asserts{{
		diagtest.IsError,
		diagtest.DetailContains("github api failed"),
	}}.AssertMatch(s.T(), diags, nil)
```

`diagtest.Asserts` already checks the length, and `DetailContains` would be a less fragile way to check that api error is present _in some form_ inside the diagnostic

Reply:
I agree, Thanks